go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -319,48 +327,63 @@ func switchAndFetchReplicaHead(ctx context.Context, branch string, db sql.Databa
 	}
 
 	// check whether branch is on remote before creating local tracking branch
-	cm, err := actions.FetchRemoteBranch(ctx, destDb.tmpDir, destDb.remote, destDb.srcDB, destDb.DbData().Ddb, branchRef, nil, actions.NoopRunProgFuncs, actions.NoopStopProgFuncs)
+	cm, err := actions.FetchRemoteBranch(ctx, db.tmpDir, db.remote, db.srcDB, db.DbData().Ddb, branchRef, actions.NoopRunProgFuncs, actions.NoopStopProgFuncs)
 	if err != nil {
 		return err
 	}
 
 	// create refs/heads/branch dataset
 	if !branchExists {
-		err = destDb.ddb.NewBranchAtCommit(ctx, branchRef, cm)
+		err = db.ddb.NewBranchAtCommit(ctx, branchRef, cm)
 		if err != nil {
 			return err
 		}
 	}
 
 	// update ReadReplicaRemote with new HEAD
 	// dolt_replicate_heads configuration remains unchanged
-	destDb, err = destDb.SetHeadRef(branchRef)
+	db, err = db.SetHeadRef(branchRef)
 	if err != nil {
 		return err
 	}
 
 	// create workingSets/heads/branch and update the working set
-	err = pullBranches(ctx, destDb, []string{branch})
+	err = pullBranches(ctx, db, []string{branch})
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func isBranch(ctx context.Context, ddb *doltdb.DoltDB, revSpec string) bool {
-	branches, err := ddb.GetBranches(ctx)
-	if err != nil {
-		return false
+// isBranch returns whether a branch with the given name is in scope for the database given
+func isBranch(ctx context.Context, db SqlDatabase, branchName string) (bool, error) {
+	var ddbs []*doltdb.DoltDB
+
+	if rdb, ok := db.(ReadReplicaDatabase); ok {
+		remoteDB, err := rdb.remote.GetRemoteDB(ctx, rdb.ddb.Format())
+		if err != nil {
+			return false, err
+		}
+		ddbs = append(ddbs, rdb.ddb, remoteDB)
```

Comment:
this looks like an addition? you added a test for getting a commit hash, what's the scenario that hits this new logic?